crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -217,15 +217,27 @@ pub unsafe trait SystemParam: Sized {
     #[allow(unused_variables)]
     fn queue(state: &mut Self::State, system_meta: &SystemMeta, world: DeferredWorld) {}
 
-    /// Creates a parameter to be passed into a [`SystemParamFunction`].
+    /// Validates that the data can be acquired by [`get_param`](SystemParam::get_param).
+    /// For systems this means they won't be executed.
     ///
-    /// [`SystemParamFunction`]: super::SystemParamFunction
+    /// # Safety
+    ///
+    /// - The passed [`UnsafeWorldCell`] must have read access to any world data
+    ///   registered in [`init_state`](SystemParam::init_state).
+    /// - `world` must be the same [`World`] that was used to initialize [`state`](SystemParam::init_state).
+    unsafe fn validate_param(
```

Comment:
Do we have docs anywhere that the implementor is only allowed to read the registered data + world metadata and not allowed to mutate any data through the `UnsafeWorldCell`

Reply:
No, I'll add that